src/Libraries/PythonNodeModelsWpf/IronPythonCompletionProvider.cs

```diff
@@ -47,6 +44,38 @@ public ScriptScope Scope
             set { scope = value; }
         }
 
+        /// <summary>
+        /// Cache storing a reference to the Dynamo Core directory
+        /// </summary>
+        private static string dynamoCoreDir { get; set; }
+
+        /// <summary>
+        /// Get the location of the Python Standard Library
+        /// </summary>
+        private static string pythonLibDir
+        {
+            get
+            {
+                // Attempt to get and cache the Dynamo Core directory path
+                if (string.IsNullOrEmpty(dynamoCoreDir))
+                {
+                    // Gather executing assembly location
+                    string nodePath = Assembly.GetExecutingAssembly().Location;
```

Comment:
this seems odd - I can see this returning odd results on some systems, could it be worth trying to get access to the pathManager or pathResolver which has all of these paths?

Reply:
This class is initialized in ScriptEditor, you can pass in `dynamoViewModel.Model.PathManager.DynamoCoreDirectory` when initializing instead of holding on this new property here, it is a dup of what we have in `PathManager`.

Reply:
^awesome, thanks @QilongTang!

Reply:
@mjkkirschner I was thinking that's why I would need to duplicate everything from the original since it can't be called from within the new constructor but I wasn't sure

Reply:
ahh, I see, it's a big constructor, yeah you can use chaining to call the other constructor with the `:this() ` syntax.